plans/objchange/objchange.go

```diff
@@ -118,165 +122,169 @@ func proposedNewObject(schema *configschema.Block, prior, config cty.Value) cty.
 	for name, blockType := range schema.BlockTypes {
 		priorV := prior.GetAttr(name)
 		configV := config.GetAttr(name)
-		var newV cty.Value
-		switch blockType.Nesting {
+		newAttrs[name] = proposedNewNestedBlock(blockType, priorV, configV)
+	}
 
-		case configschema.NestingSingle, configschema.NestingGroup:
-			newV = ProposedNewObject(&blockType.Block, priorV, configV)
+	return cty.ObjectVal(newAttrs)
+}
+
+func proposedNewNestedBlock(schema *configschema.NestedBlock, prior, config cty.Value) cty.Value {
```

Comment:
The code used in here may still contain a lot of assumption from blocks, which are always a zero value rather than `null`. We may need to refactor this further to split out attr vs block behavior

Reply:
I did that thing, thank you